src/DataGrid.tsx

```diff
@@ -438,22 +444,23 @@ function DataGrid<R, SR, K extends Key>(
   });
 
   useLayoutEffect(() => {
-    if (!isWidthInitialized || flexWidthViewportColumns.length === 0) return;
-
-    setColumnWidths((columnWidths) => {
-      const newColumnWidths = new Map(columnWidths);
-      const grid = gridRef.current!;
-
-      for (const column of flexWidthViewportColumns) {
-        const measuringCell = grid.querySelector(`[data-measuring-cell-key="${column.key}"]`)!;
-        // Set the actual width of the column after it is rendered
-        const { width } = measuringCell.getBoundingClientRect();
-        newColumnWidths.set(column.key, width);
+    const newMeasuredColumnWidths = new Map(measuredColumnWidths);
+    for (const { key, width } of viewportColumns) {
+      if (
+        !resizedColumnWidths.has(key) &&
+        !newMeasuredColumnWidths.has(key) &&
+        typeof width === 'string'
+      ) {
+        const measuringCell = gridRef.current!.querySelector(`[data-measuring-cell-key="${key}"]`)!;
+        const measuredWidth = measuringCell.getBoundingClientRect().width;
+        newMeasuredColumnWidths.set(key, measuredWidth);
       }
+    }
 
-      return newColumnWidths;
-    });
-  }, [isWidthInitialized, flexWidthViewportColumns, gridRef]);
+    if (newMeasuredColumnWidths.size !== measuredColumnWidths.size) {
```

Comment:
This doesn't seem like a safe check, should we explicitly check if we found an updated/new width in the loop?